streams/src/main/java/org/apache/kafka/streams/state/internals/metrics/RocksDBMetricsRecordingTrigger.java

```diff
@@ -32,12 +32,12 @@ public RocksDBMetricsRecordingTrigger(final Time time) {
 
     public void addMetricsRecorder(final RocksDBMetricsRecorder metricsRecorder) {
         final String metricsRecorderName = metricsRecorderName(metricsRecorder);
-        if (metricsRecordersToTrigger.containsKey(metricsRecorderName)) {
+        final RocksDBMetricsRecorder existingRocksDBMetricsRecorder = metricsRecordersToTrigger.putIfAbsent(metricsRecorderName, metricsRecorder);
```

Comment:
Does this change behavior? Potential concurrency bug aside (which I'm not sure is valid, since it's unclear if we expect this method to be called concurrently), it looks like we're going from failing _before_ overwriting values to now failing _after_ overwriting them. Is there any fallout from that or is it a benign change?